src/twisted/conch/test/test_userauth.py

```diff
@@ -390,6 +406,337 @@ def check(ignored):
 
         return d.addCallback(check)
 
+    def test_verifyValidPrivateKey_sk_ed25519(self):
+        """
+        Test that verifying a valid sk-ssh-ed25519 public key works.
+        """
+        sk = DummySK()
+        application_name = "test-application"
+        enroll = sk.enroll(
+            SKAlgorithm.ED25519,
+            challenge=b"dummy-challenge",
+            application=application_name,
+            flags=0x01,  # user presence
+        )
+
+        alg_name = b"sk-ssh-ed25519@openssh.com"
+
+        # string alg-name
+        # string raw-public-key
+        # string application ("ssh:")
+        pubkey_blob = NS(alg_name) + NS(enroll.public_key) + NS(b"ssh:")
+
+        packet = (
+            NS(b"foo")
+            + NS(b"none")
+            + NS(b"publickey")
+            + b"\x00"
+            + NS(alg_name)
+            + NS(pubkey_blob)
+        )
+
+        d = self.authServer.ssh_USERAUTH_REQUEST(packet)
```

Comment:
Minor comment

Will this code spin the reactor ?

If not, we can extract the result without needed the `addCallback` ... should make the test a bit easier to read.

```suggestion
        d = self.authServer.ssh_USERAUTH_REQUEST(packet)
        self.successResultOf(d)
```

Reply:
This seems to work. Do you want me to update all tests like this? This would also involve refactoring some related code like `_checkFailed` etc.

Reply:
If it's a lot of work, we can keep this code.

The problem is that when we return a deferrd, it implies that the test will do some async I/O operations

When using `self.successResultOf(d)` in a test, we also validate that the code is `blocking`

Reply:
I think changing this everywhere would make this PR a bit convoluted, but I would be happy to submit a PR next week to refactor all tests with this pattern :+1:

Reply:
We can leave it like this for now, and spend the time on more productive tasks

see #12496  #12184 #12217 or even #12139

I can help with some of the fixes, but I need someone to help with the review